sekoia_automation/connector/event_aggregator.py

```diff
@@ -0,0 +1,187 @@
+import copy
+import datetime
+import threading
+import time
+from typing import Callable, Tuple
+
+import xxhash as xxhash
+from ciso8601 import parse_datetime_as_naive
+from pydantic import BaseModel
+
+
+class Fingerprint(BaseModel):
+    condition_func: Callable[[dict], bool]
+    build_hash_str_func: Callable[[dict], str]
```

Comment:
Maybe we could have a single function that, if it returns `None`, will omit aggregation ?

The idea behind it is to avoid to have to iterate the events twice, once to check if it can be aggregated and a second time to calculate the fingerprint.

```suggestion
    build_hash_str_func: Callable[[dict], str | None]
```

Reply:
applied